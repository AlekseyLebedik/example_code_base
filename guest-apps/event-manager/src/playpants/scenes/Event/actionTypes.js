const prefix = 'playpants/EVENT';
export const CHANGES = `${prefix}.CHANGES`;
export const CLEAR_EVENT_STORE = `${prefix}.CLEAR_EVENT_STORE`;
export const CREATE_COMMENT = `${prefix}.COMMENT`;
export const DELETE_EVENT = `${prefix}.DELETE_EVENT`;
export const DISCUSSION_FETCH = `${prefix}.DISCUSSION`;
export const EVENT_FETCH = `${prefix}.EVENT`;
export const FETCH_USER_RESPONSIBILITIES = `${prefix}.USER_RESPONSIBILITIES`;
export const UNLOCK_THEN_CLEAR = `${prefix}.UNLOCK_THEN_CLEAR`;
export const UNLOCK_THEN_DELETE = `${prefix}.UNLOCK_THEN_DELETE`;
export const UPDATE_AUTHS = `${prefix}.AUTHS`;
export const UPDATE_EVENT = `${prefix}.EVENT`;
