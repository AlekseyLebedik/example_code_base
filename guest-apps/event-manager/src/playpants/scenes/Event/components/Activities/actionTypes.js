const prefix = 'playpants/ACTIVITIES';

export const CHANGE_ACTIVITIES_TYPE = `${prefix}.CHANGE_ACTIVITIES_TYPE`;
export const ACTIVITIES_SEARCH = `${prefix}.ACTIVITIES_SEARCH`;

export const CREATE_ACTIVITY = `${prefix}.ACTIVITY_CREATE`;
export const CREATE_ACTIVITY_FAILED = `${prefix}.ACTIVITY_CREATE_FAILED`;
export const CREATE_ACTIVITY_SUCCESS = `${prefix}.ACTIVITY_CREATE_SUCCESS`;

export const DUPLICATE_ACTIVITY = `${prefix}.ACTIVITY_DUPLICATE`;
export const REVERT_ACTIVITY = `${prefix}.ACTIVITY_REVERT`;

export const DELETE_ACTIVITY = `${prefix}.ACTIVITY_DELETE`;
export const DELETE_ACTIVITY_SUCCESS = `${prefix}.ACTIVITY_DELETE_SUCCESS`;
export const DELETE_ACTIVITY_FAILED = `${prefix}.ACTIVITY_DELETE_FAILED`;

export const UPDATE_ACTIVITY = `${prefix}.ACTIVITY_UPDATE`;
export const UPDATE_ACTIVITY_SUCCESS = `${prefix}.ACTIVITY_UPDATE_SUCCESS`;
export const UPDATE_ACTIVITY_FAILED = `${prefix}.ACTIVITY_UPDATE_FAILED`;

export const UPDATE_THEN_REVERT_ACTIVITY = `${prefix}.ACTIVITY_UPDATE_THEN_REVERT`;

export const TITLE_ENV_FETCH = `${prefix}.TITLE_ENV`;
export const CONTEXTS_FETCH = `${prefix}.CONTEXTS`;
