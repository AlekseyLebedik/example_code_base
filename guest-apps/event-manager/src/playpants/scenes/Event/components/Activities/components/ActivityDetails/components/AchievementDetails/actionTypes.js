const prefix = 'playpants/ACTIVITIES.AE';

export const FETCH_RULESET_LIST = `${prefix}.RULESET_LIST`;
export const FETCH_SELECTED_RULESET = `${prefix}.SELECTED_RULESET`;
export const CLEAR_ACHIEVEMENTS = `${prefix}.CLEAR_ACHIEVEMENTS`;
export const CLEAR_RULESET = `${prefix}.CLEAR_RULESET`;
