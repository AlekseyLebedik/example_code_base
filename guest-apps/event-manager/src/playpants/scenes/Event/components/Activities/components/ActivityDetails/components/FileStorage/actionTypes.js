const prefix = 'playpants/FILESTORAGE';

export const UPLOAD_FILE = `${prefix}.FILE_UPLOAD`;
export const UPLOAD_FILE_SUCCESS = `${prefix}.FILE_UPLOAD_SUCCESS`;
export const UPLOAD_FILE_FAILED = `${prefix}.FILE_UPLOAD_FAILED`;

export const UPLOAD_PROGRESS_FETCH = `${prefix}.UPLOAD_PROGRESS_FETCH`;
export const UPLOAD_PROGRESS_FETCH_SUCCESS = `${prefix}.UPLOAD_PROGRESS_FETCH_SUCCESS`;
export const UPLOAD_PROGRESS_FETCH_FAILED = `${prefix}.UPLOAD_PROGRESS_FETCH_FAILED`;

export const DOWNLOAD_FILE = `${prefix}.FILE_DOWNLOAD`;
export const DOWNLOAD_FILE_SUCCESS = `${prefix}.FILE_DOWNLOAD_SUCCESS`;
export const DOWNLOAD_FILE_FAILED = `${prefix}.FILE_DOWNLOAD_FAILED`;

export const REMOVE_FILE = `${prefix}.FILE_DELETE`;
export const REMOVE_FILE_SUCCESS = `${prefix}.FILE_DELETE_SUCCESS`;
export const REMOVE_FILE_FAILED = `${prefix}.FILE_DELETE_FAILED`;

export const UPDATE_FILE = `${prefix}.FILE_UPDATE`;
export const UPDATE_FILE_SUCCESS = `${prefix}.FILE_UPDATE_SUCCESS`;
export const UPDATE_FILE_FAILED = `${prefix}.FILE_UPDATE_FAILED`;

export const FILE_DETAILS_FETCH = `${prefix}.FILE_DETAILS_FETCH`;
export const FILE_DETAILS_FETCH_SUCCESS = `${prefix}.FILE_DETAILS_FETCH_SUCCESS`;
export const FILE_DETAILS_FETCH_FAILED = `${prefix}.FILE_DETAILS_FETCH_FAILED`;

export const FILESTORAGE_CONTEXTS_FETCH = `${prefix}.FILESTORAGE_CONTEXTS`;
