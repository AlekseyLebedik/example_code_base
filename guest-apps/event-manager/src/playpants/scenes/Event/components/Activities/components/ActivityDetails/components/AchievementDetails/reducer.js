import { combineReducers } from 'redux';
import { createFetchReducer } from '@demonware/devzone-core/helpers/reducers';
import reduceReducers from 'reduce-reducers';
import * as AT from './actionTypes';

const INITIAL_STATE = {
  rulesetList: {
    data: [],
    loading: false,
  },
  selectedRuleset: {
    data: {},
    loading: false,
  },
};

const rulesetListFetchReducer = createFetchReducer(AT.FETCH_RULESET_LIST);

const selectedRulesetFetchReducer = createFetchReducer(
  AT.FETCH_SELECTED_RULESET
);

const rulesetListReducer = (state = INITIAL_STATE.rulesetList, action) => {
  switch (action.type) {
    case `${AT.FETCH_RULESET_LIST}_FETCH_FAILED`:
      return {
        ...INITIAL_STATE.rulesetList,
      };
    default:
      return {
        ...rulesetListFetchReducer(state, action),
      };
  }
};

const selectedRulesetReducer = (
  state = INITIAL_STATE.selectedRuleset,
  action
) => {
  switch (action.type) {
    case `${AT.FETCH_SELECTED_RULESET}_FETCH_FAILED`:
      return {
        ...INITIAL_STATE.selectedRuleset,
      };
    default:
      return {
        ...selectedRulesetFetchReducer(state, action),
      };
  }
};

const clearAchievementsReducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case AT.CLEAR_ACHIEVEMENTS: {
      return {
        ...INITIAL_STATE,
      };
    }
    case AT.CLEAR_RULESET: {
      return {
        ...state,
        selectedRuleset: { ...INITIAL_STATE.selectedRuleset },
      };
    }
    default:
      return state;
  }
};

const reducer = reduceReducers(
  clearAchievementsReducer,
  combineReducers({
    rulesetList: rulesetListReducer,
    selectedRuleset: selectedRulesetReducer,
  })
);

export default reducer;
