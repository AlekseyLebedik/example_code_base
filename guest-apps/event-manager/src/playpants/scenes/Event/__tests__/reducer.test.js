import mockState from 'playpants/testUtils/mockState';
import reducer from '../reducer';
import * as AT from '../actionTypes';

const { discussion } = mockState.Scenes.Event;

const testEvent = {
  id: 5,
  title: 'asll',
  created_at: 1566330583,
  created_by: { id: 1, name: 'Initial User' },
  publish_at: 1566590400,
  end_at: null,
  status: 'open',
  activities: [
    {
      id: 11,
      type: 'pubvars',
      activity: '{"variable_sets":[]}',
      title_envs: [1],
      publish_on: 'on_start',
      exec_order: 0,
      updated_by: { id: 1, name: 'Initial User' },
    },
  ],
  updated_at: 1566335105,
  updated_by: { id: 1, name: 'Initial User' },
  note: '',
  project: 1,
  env_type: 'Development',
  authorizations: [],
  locked_by: null,
  task: null,
  manual_tags: '[]',
  auto_tags: '[]',
  platforms: '["CROSSPLAY","PC"]',
  authorizers: [
    { id: 1, name: 'Initial User' },
    { id: 11, name: 'Admin' },
  ],
};

describe('Event Reducer', () => {
  it('returns default state', () => {
    const state = reducer(undefined, {});
    expect(state).toMatchSnapshot();
  });

  describe('EVENT_FETCH', () => {
    it('fetches an event successfully', () => {
      const action = {
        type: `${AT.EVENT_FETCH}_FETCH_SUCCESS`,
        data: testEvent,
      };

      const state = reducer(undefined, action);
      expect(state).toMatchSnapshot();
    });
  });

  describe('UPDATE_EVENT', () => {
    it('attempts to edit an event', () => {
      const action = {
        type: `${AT.UPDATE_EVENT}_UPDATE`,
        data: {},
      };

      const state = reducer(undefined, action);
      expect(state).toMatchSnapshot();
    });
  });

  describe('UPDATE_EVENT_SUCCEED', () => {
    it('returned the relevant data for a successful edit of the event', () => {
      const action = {
        type: `${AT.UPDATE_EVENT}_UPDATE_SUCCESS`,
        data: testEvent,
      };

      const state = reducer(undefined, action);
      expect(state).toMatchSnapshot();
    });
  });

  describe('DISCUSSION_FETCH_SUCCEED', () => {
    it('changes the activity type', () => {
      const action = {
        type: `${AT.DISCUSSION_FETCH}_FETCH_SUCCESS`,
        discussion,
      };

      const state = reducer(undefined, action);
      expect(state).toMatchSnapshot();
    });
  });

  describe('CLEAR_CHANGE_STATUS', () => {
    it('clears change status for activity and event', () => {
      const action = {
        type: AT.CLEAR_CHANGE_STATUS,
      };
      const state = reducer(undefined, action);
      expect(state).toMatchSnapshot();
    });
  });
});
