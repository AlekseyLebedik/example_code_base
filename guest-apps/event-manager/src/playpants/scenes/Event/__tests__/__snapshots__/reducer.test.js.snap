// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Reducer CLEAR_CHANGE_STATUS clears change status for activity and event 1`] = `
Object {
  "activity": Object {
    "achievements": Object {
      "rulesetList": Object {
        "data": Array [],
        "loading": false,
      },
      "selectedRuleset": Object {
        "data": Object {},
        "loading": false,
      },
    },
    "contexts": Object {
      "data": Array [],
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
    "envSettings": Object {
      "data": Array [],
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
    "filestorage": Object {
      "contexts": Object {
        "data": Array [],
        "error": false,
        "loading": false,
        "next": undefined,
        "nextPageToken": undefined,
        "params": undefined,
      },
      "files": Object {},
      "uploadProgress": Object {},
    },
    "publisherObjects": Object {
      "categories": Object {
        "data": Array [],
        "loading": false,
      },
      "error": Object {},
      "groups": Object {
        "data": Array [],
        "loading": false,
      },
    },
    "pubvars": Object {
      "selectedValues": Object {
        "context": null,
        "group_id": null,
        "namespace": null,
      },
      "variableSets": Array [],
    },
    "pyscripts": Object {
      "templates": Object {
        "data": Array [],
        "error": false,
        "loading": false,
        "next": undefined,
        "nextPageToken": undefined,
        "params": undefined,
      },
    },
    "searchAvailable": true,
    "selectedActivityType": "all",
    "thunderpants": Object {
      "buildList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "buildSchema": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "deployerList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "deploymentList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "filterType": "default",
      "form": Object {
        "data": Object {},
        "next": "",
        "schema": Array [],
        "summary": Object {},
        "targets": Array [],
        "type": "",
      },
      "targetList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "unlockedDeployments": Object {},
      "userParamsSchema": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "viewList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
    },
  },
  "conflicts": Object {
    "conflictList": Array [],
    "conflictType": "all",
    "query": "",
  },
  "discussion": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "event": Object {
    "data": Object {},
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "history": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "responsibilities": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "taskMonitor": Object {
    "taskDetail": Object {
      "data": Object {},
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
    "tasks": Object {
      "data": Array [],
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
  },
}
`;

exports[`Event Reducer DISCUSSION_FETCH_SUCCEED changes the activity type 1`] = `
Object {
  "activity": Object {
    "achievements": Object {
      "rulesetList": Object {
        "data": Array [],
        "loading": false,
      },
      "selectedRuleset": Object {
        "data": Object {},
        "loading": false,
      },
    },
    "contexts": Object {
      "data": Array [],
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
    "envSettings": Object {
      "data": Array [],
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
    "filestorage": Object {
      "contexts": Object {
        "data": Array [],
        "error": false,
        "loading": false,
        "next": undefined,
        "nextPageToken": undefined,
        "params": undefined,
      },
      "files": Object {},
      "uploadProgress": Object {},
    },
    "publisherObjects": Object {
      "categories": Object {
        "data": Array [],
        "loading": false,
      },
      "error": Object {},
      "groups": Object {
        "data": Array [],
        "loading": false,
      },
    },
    "pubvars": Object {
      "selectedValues": Object {
        "context": null,
        "group_id": null,
        "namespace": null,
      },
      "variableSets": Array [],
    },
    "pyscripts": Object {
      "templates": Object {
        "data": Array [],
        "error": false,
        "loading": false,
        "next": undefined,
        "nextPageToken": undefined,
        "params": undefined,
      },
    },
    "searchAvailable": true,
    "selectedActivityType": "all",
    "thunderpants": Object {
      "buildList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "buildSchema": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "deployerList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "deploymentList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "filterType": "default",
      "form": Object {
        "data": Object {},
        "next": "",
        "schema": Array [],
        "summary": Object {},
        "targets": Array [],
        "type": "",
      },
      "targetList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "unlockedDeployments": Object {},
      "userParamsSchema": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "viewList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
    },
  },
  "conflicts": Object {
    "conflictList": Array [],
    "conflictType": "all",
    "query": "",
  },
  "discussion": Object {
    "data": undefined,
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "event": Object {
    "data": Object {},
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "history": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "responsibilities": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "taskMonitor": Object {
    "taskDetail": Object {
      "data": Object {},
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
    "tasks": Object {
      "data": Array [],
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
  },
}
`;

exports[`Event Reducer EVENT_FETCH fetches an event successfully 1`] = `
Object {
  "activity": Object {
    "achievements": Object {
      "rulesetList": Object {
        "data": Array [],
        "loading": false,
      },
      "selectedRuleset": Object {
        "data": Object {},
        "loading": false,
      },
    },
    "contexts": Object {
      "data": Array [],
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
    "envSettings": Object {
      "data": Array [],
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
    "filestorage": Object {
      "contexts": Object {
        "data": Array [],
        "error": false,
        "loading": false,
        "next": undefined,
        "nextPageToken": undefined,
        "params": undefined,
      },
      "files": Object {},
      "uploadProgress": Object {},
    },
    "publisherObjects": Object {
      "categories": Object {
        "data": Array [],
        "loading": false,
      },
      "error": Object {},
      "groups": Object {
        "data": Array [],
        "loading": false,
      },
    },
    "pubvars": Object {
      "selectedValues": Object {
        "context": null,
        "group_id": null,
        "namespace": null,
      },
      "variableSets": Array [],
    },
    "pyscripts": Object {
      "templates": Object {
        "data": Array [],
        "error": false,
        "loading": false,
        "next": undefined,
        "nextPageToken": undefined,
        "params": undefined,
      },
    },
    "searchAvailable": true,
    "selectedActivityType": "all",
    "thunderpants": Object {
      "buildList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "buildSchema": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "deployerList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "deploymentList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "filterType": "default",
      "form": Object {
        "data": Object {},
        "next": "",
        "schema": Array [],
        "summary": Object {},
        "targets": Array [],
        "type": "",
      },
      "targetList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "unlockedDeployments": Object {},
      "userParamsSchema": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "viewList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
    },
  },
  "conflicts": Object {
    "conflictList": Array [],
    "conflictType": "all",
    "query": "",
  },
  "discussion": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "event": Object {
    "data": Object {
      "activities": Array [
        Object {
          "activity": Object {
            "variable_sets": Array [],
          },
          "exec_order": 0,
          "id": 11,
          "publish_on": "on_start",
          "title_envs": Array [
            1,
          ],
          "type": "pubvars",
          "updated_by": Object {
            "id": 1,
            "name": "Initial User",
          },
        },
      ],
      "authorizations": Array [],
      "authorizers": Array [
        Object {
          "id": 1,
          "name": "Initial User",
        },
        Object {
          "id": 11,
          "name": "Admin",
        },
      ],
      "auto_tags": Array [],
      "created_at": 1566330583,
      "created_by": Object {
        "id": 1,
        "name": "Initial User",
      },
      "end_at": null,
      "env_type": "Development",
      "id": 5,
      "locked_by": null,
      "manual_tags": Array [],
      "note": "",
      "platforms": Array [
        "CROSSPLAY",
        "PC",
      ],
      "project": 1,
      "publish_at": 1566590400,
      "status": "open",
      "task": null,
      "title": "asll",
      "updated_at": 1566335105,
      "updated_by": Object {
        "id": 1,
        "name": "Initial User",
      },
    },
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "history": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "responsibilities": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "taskMonitor": Object {
    "taskDetail": Object {
      "data": Object {},
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
    "tasks": Object {
      "data": Array [],
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
  },
}
`;

exports[`Event Reducer UPDATE_EVENT attempts to edit an event 1`] = `
Object {
  "activity": Object {
    "achievements": Object {
      "rulesetList": Object {
        "data": Array [],
        "loading": false,
      },
      "selectedRuleset": Object {
        "data": Object {},
        "loading": false,
      },
    },
    "contexts": Object {
      "data": Array [],
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
    "envSettings": Object {
      "data": Array [],
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
    "filestorage": Object {
      "contexts": Object {
        "data": Array [],
        "error": false,
        "loading": false,
        "next": undefined,
        "nextPageToken": undefined,
        "params": undefined,
      },
      "files": Object {},
      "uploadProgress": Object {},
    },
    "publisherObjects": Object {
      "categories": Object {
        "data": Array [],
        "loading": false,
      },
      "error": Object {},
      "groups": Object {
        "data": Array [],
        "loading": false,
      },
    },
    "pubvars": Object {
      "selectedValues": Object {
        "context": null,
        "group_id": null,
        "namespace": null,
      },
      "variableSets": Array [],
    },
    "pyscripts": Object {
      "templates": Object {
        "data": Array [],
        "error": false,
        "loading": false,
        "next": undefined,
        "nextPageToken": undefined,
        "params": undefined,
      },
    },
    "searchAvailable": true,
    "selectedActivityType": "all",
    "thunderpants": Object {
      "buildList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "buildSchema": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "deployerList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "deploymentList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "filterType": "default",
      "form": Object {
        "data": Object {},
        "next": "",
        "schema": Array [],
        "summary": Object {},
        "targets": Array [],
        "type": "",
      },
      "targetList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "unlockedDeployments": Object {},
      "userParamsSchema": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "viewList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
    },
  },
  "conflicts": Object {
    "conflictList": Array [],
    "conflictType": "all",
    "query": "",
  },
  "discussion": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "event": Object {
    "data": Object {},
    "error": false,
    "loading": true,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "history": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "responsibilities": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "taskMonitor": Object {
    "taskDetail": Object {
      "data": Object {},
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
    "tasks": Object {
      "data": Array [],
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
  },
}
`;

exports[`Event Reducer UPDATE_EVENT_SUCCEED returned the relevant data for a successful edit of the event 1`] = `
Object {
  "activity": Object {
    "achievements": Object {
      "rulesetList": Object {
        "data": Array [],
        "loading": false,
      },
      "selectedRuleset": Object {
        "data": Object {},
        "loading": false,
      },
    },
    "contexts": Object {
      "data": Array [],
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
    "envSettings": Object {
      "data": Array [],
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
    "filestorage": Object {
      "contexts": Object {
        "data": Array [],
        "error": false,
        "loading": false,
        "next": undefined,
        "nextPageToken": undefined,
        "params": undefined,
      },
      "files": Object {},
      "uploadProgress": Object {},
    },
    "publisherObjects": Object {
      "categories": Object {
        "data": Array [],
        "loading": false,
      },
      "error": Object {},
      "groups": Object {
        "data": Array [],
        "loading": false,
      },
    },
    "pubvars": Object {
      "selectedValues": Object {
        "context": null,
        "group_id": null,
        "namespace": null,
      },
      "variableSets": Array [],
    },
    "pyscripts": Object {
      "templates": Object {
        "data": Array [],
        "error": false,
        "loading": false,
        "next": undefined,
        "nextPageToken": undefined,
        "params": undefined,
      },
    },
    "searchAvailable": true,
    "selectedActivityType": "all",
    "thunderpants": Object {
      "buildList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "buildSchema": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "deployerList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "deploymentList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "filterType": "default",
      "form": Object {
        "data": Object {},
        "next": "",
        "schema": Array [],
        "summary": Object {},
        "targets": Array [],
        "type": "",
      },
      "targetList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "unlockedDeployments": Object {},
      "userParamsSchema": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "viewList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
    },
  },
  "conflicts": Object {
    "conflictList": Array [],
    "conflictType": "all",
    "query": "",
  },
  "discussion": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "event": Object {
    "data": Object {
      "activities": Array [
        Object {
          "activity": Object {
            "variable_sets": Array [],
          },
          "exec_order": 0,
          "id": 11,
          "publish_on": "on_start",
          "title_envs": Array [
            1,
          ],
          "type": "pubvars",
          "updated_by": Object {
            "id": 1,
            "name": "Initial User",
          },
        },
      ],
      "authorizations": Array [],
      "authorizers": Array [
        Object {
          "id": 1,
          "name": "Initial User",
        },
        Object {
          "id": 11,
          "name": "Admin",
        },
      ],
      "auto_tags": Array [],
      "created_at": 1566330583,
      "created_by": Object {
        "id": 1,
        "name": "Initial User",
      },
      "end_at": null,
      "env_type": "Development",
      "id": 5,
      "locked_by": null,
      "manual_tags": Array [],
      "note": "",
      "platforms": Array [
        "CROSSPLAY",
        "PC",
      ],
      "project": 1,
      "publish_at": 1566590400,
      "status": "open",
      "task": null,
      "title": "asll",
      "updated_at": 1566335105,
      "updated_by": Object {
        "id": 1,
        "name": "Initial User",
      },
    },
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "history": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "responsibilities": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "taskMonitor": Object {
    "taskDetail": Object {
      "data": Object {},
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
    "tasks": Object {
      "data": Array [],
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
  },
}
`;

exports[`Event Reducer returns default state 1`] = `
Object {
  "activity": Object {
    "achievements": Object {
      "rulesetList": Object {
        "data": Array [],
        "loading": false,
      },
      "selectedRuleset": Object {
        "data": Object {},
        "loading": false,
      },
    },
    "contexts": Object {
      "data": Array [],
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
    "envSettings": Object {
      "data": Array [],
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
    "filestorage": Object {
      "contexts": Object {
        "data": Array [],
        "error": false,
        "loading": false,
        "next": undefined,
        "nextPageToken": undefined,
        "params": undefined,
      },
      "files": Object {},
      "uploadProgress": Object {},
    },
    "publisherObjects": Object {
      "categories": Object {
        "data": Array [],
        "loading": false,
      },
      "error": Object {},
      "groups": Object {
        "data": Array [],
        "loading": false,
      },
    },
    "pubvars": Object {
      "selectedValues": Object {
        "context": null,
        "group_id": null,
        "namespace": null,
      },
      "variableSets": Array [],
    },
    "pyscripts": Object {
      "templates": Object {
        "data": Array [],
        "error": false,
        "loading": false,
        "next": undefined,
        "nextPageToken": undefined,
        "params": undefined,
      },
    },
    "searchAvailable": true,
    "selectedActivityType": "all",
    "thunderpants": Object {
      "buildList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "buildSchema": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "deployerList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "deploymentList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "filterType": "default",
      "form": Object {
        "data": Object {},
        "next": "",
        "schema": Array [],
        "summary": Object {},
        "targets": Array [],
        "type": "",
      },
      "targetList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "unlockedDeployments": Object {},
      "userParamsSchema": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
      "viewList": Object {
        "data": Array [],
        "loading": true,
        "success": false,
      },
    },
  },
  "conflicts": Object {
    "conflictList": Array [],
    "conflictType": "all",
    "query": "",
  },
  "discussion": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "event": Object {
    "data": Object {},
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "history": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "responsibilities": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "taskMonitor": Object {
    "taskDetail": Object {
      "data": Object {},
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
    "tasks": Object {
      "data": Array [],
      "error": false,
      "loading": false,
      "next": undefined,
      "nextPageToken": undefined,
      "params": undefined,
    },
  },
}
`;
